src/win/error.c

```diff
@@ -141,7 +141,7 @@ int uv_translate_sys_error(int sys_errno) {
     case ERROR_MOD_NOT_FOUND:               return UV_ENOENT;
     case ERROR_PATH_NOT_FOUND:              return UV_ENOENT;
     case WSAHOST_NOT_FOUND:                 return UV_ENOENT;
-    case WSANO_DATA:                        return UV_ENOENT;
+    case WSANO_DATA:                        return UV_EADDRNOTAVAIL;
```

Comment:
I don't think `UV_EADDRNOTAVAIL` is the appropriate mapping for `WSANO_DATA` (although I can understand how you arrived at it - "no address for hostname") but what's more, I worry it's going to break existing programs.

(Might be an academic fear because I don't think DNS errors go through this function but better safe than sorry.)